external-crates/move/crates/move-package/src/resolution/dependency_graph.rs

```diff
@@ -201,8 +202,10 @@ impl<Progress: Write> DependencyGraphBuilder<Progress> {
         // compute digests eagerly as even if we can't reuse existing lock file, they need to become
         // part of the newly computed dependency graph
         let new_manifest_digest = digest_str(manifest_string.into_bytes().as_slice());
+        let lock_path = root_path.join(SourcePackageLayout::Lock.path());
+        let mut lock_file = File::open(lock_path)?;
         let (old_manifest_digest_opt, old_deps_digest_opt, lock_string) = match lock_string_opt {
-            Some(lock_string) => match schema::read_header(&lock_string) {
+            Some(lock_string) => match schema::Header::read(&mut lock_file) {
```

Comment:
This is the only place we did `schema:read_header` outside of the lock file schema, and now it uses `schema::Header::read`, which now becomes the canonical public interface for reading the lock file header.